OfflineAudioContext/explainer.md

```diff
@@ -0,0 +1,247 @@
+
+### Non-goals
+
+- Change the existing `startRendering()` behavior, this API change is additive
+
+## Proposed Approach - Add `startRenderingStream()` function
+
+The preferred approach is adding a new method `startRenderingStream()` that yields buffers of interleaved audio samples in a Float32Array, or another format as outlined in Open Questions. In this scenario, the user can read chunks as they arrive and consume them for storage, transcoding via WebCodecs, sending to a server, etc.
+
+Usage example:
+
+```js
+const context = new OfflineAudioContext({ numberOfChannels: 2, length: 44100, sampleRate: 44100 });
+
+// Add some nodes to build a graph...
+
+if ("startRendering" in OfflineAudioContext) {
+  const reader = context.startRenderingStream({ format: "f32-interleaved" }).getReader(); 
+  const isRenderingCompleted = false;
+  while (!isRenderingCompleted) {
+    // get the next chunk of data from the stream
+    const result = await reader.read();
+
+    // the reader returns done = true when there are no more chunks to consume
+    if (result.done) {
+      break;
+    }
+
+    // result.value contains interleaved Float32Array values
+    const buffers = result.value;
+  }
+} else {
+  audioBuffer = await offlineAudioContext.startRendering();
+}
+```
+
+Proposed interface:
+
+```js
+// From https://developer.mozilla.org/en-US/docs/Web/API/AudioData/format
+enum AudioFormat {
+    "u8",
+    "s16",
+    "s32",
+    "f32",
+    "u8-planar",
+    "s16-planar",
+    "s32-planar",
+    "f32-planar"
+    "f32-interleaved"
+}
+
+dictionary OfflineAudioRenderingOptions {
+    required AudioFormat format = "f32-interleaved";
+}
+
+partial interface OfflineAudioContext {
+    // Returns a stream that yields buffers of interleaved audio samples in Float32Array or whatever format is specified
+    Promise<ReadableStream> startRenderingStream(options?: OfflineAudioRenderingOptions);
+};
+```
+
+### Pros
+
+- The new capability is feature detectable because it is a new function. Compared to Alternative 1 which cannot be easily detected
+- Aligns well with other web streaming APIs, similar to [WebCodecs](https://streams.spec.whatwg.org/#readablestream)
+- Works with very large durations, no upper limit to WebAudio graph duration
+
+### Cons
+
+- Need to define sensible chunk sizes, backpressure, error handling, and end-of-stream
+
+### Output format
+
+There is an open question of what data format `startRenderingStream()` should return. The options under consideration are `AudioBuffer`, `Float32Array` planar or `Float32Array` interleaved.
+
+#### `AudioBuffer`
+
+**Pros**
+
+- semantically closest to the `startRendering()` API
+- does not add a new type to the WebAudio spec
+
+**Cons**
+
+- does not allow developers to BYOB (bring your own buffer) and BYOB helps developers manage memory usage, so `AudioBuffer` removes a bit of control
+
+#### `f32-planar` per channel
```

Comment:
More of a nitpick since it doesn't really matter. I think we should just call these options after the WebCodecs spec. Just calling them "Interleaved Float32Array" and "Planar Float32Array" makes more sense imo. Like the audio buffer is also technically `f32-planar`, it doesn't really clarify what type is returned.